### PR TITLE
Reorganize Catalog based on Catalog Organization Proposal

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

* **NOTE**: This patch covers the first part of the issue   where it modifies all the tasks and moves the tasks which don't need to be split to the task directory. The task which has to be split and moved to the task directory will be added in the next pr (https://github.com/tektoncd/catalog/pull/401) to make review easier

1. Each resource follows the following structure
```
    ./task/         👈 the kind of the resource 
      /argocd       👈 definition file must have same name
        /OWNERS     👈 owners of this ressource
        /README.md  👈 fallback README if version is missing README
        /0.1
          /README.md     👈 [optional] since there can be a fallback
          /argocd.yaml   👈 the file name must be resource name
          /samples/deploy-to-k8s.yaml
        /0.2/...
      /golang-build
        /0.1
          /README.md
          /golang-build.yaml
          /samples/golang-build.yaml
```
2. Resource YAML file includes following changes
  *  Labels include the version of the resource.
  *  Annotations include `minimum pipeline version` supported by the resource, 
     `tags` associated with the resource and `displayName` of the resource 
    
  ```yaml
    
 labels:
    app.kubernetes.io/version: "0.1"            👈 version of the resource
    
  annotations:
    tekton.dev/pipelines.minVersion: "0.12.1"   👈 will work from x version  of pipeline
    tekton.dev/tags: "ansible, cli"                        👈 Comma separated list of tags 
    tekton.dev/displayName: "Ansible Tower Cli"    
    
spec:
  description: |-
    ansible-tower-cli task simplifies 
    workflow, jobs, manage users...   👈 # Summary
    
    Ansible Tower (formerly ‘AWX’) is a ...
  
  ```

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: makisu/makisu.yaml [hunk 1]
```diff
@@ -2,43 +2,56 @@ apiVersion: tekton.dev/v1beta1
...
-  - name: CONTEXTPATH
-    description: Path to the build context.
-    default: /workspace
-  - name: PUSH_REGISTRY
-    description: Registry to push image to.
-    default: index.docker.io
-  - name: REGISTRY_SECRET
-    description: Secret containing information about the used regsitry.
-    default: docker-registry-config
+    - name: CONTEXTPATH
```
Comment (sthaha) [inline makisu/makisu.yaml:20]: this whitespace change is unrelated to the patch. could you please ensure that only necessary changes are made to the yaml files 🙏 👼 ?

**Comment by sthaha** [comment]
Please note Hub has been deployed to https://ui-tekton-hub.apps.openshift-web.p0s5.p1.openshiftapps.com/ based on this PR. I will force a resync of Hub tasks as more tasks are added to this PR.

File: github/set_status.yaml [hunk 2]
```diff
@@ -3,10 +3,26 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: github-set-status
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: github
+    displayName: "github set status"
   description: |
```
Comment (sthaha) [inline github/set_status.yaml:12]: this needs to be removed

File: github/close_issue.yaml [hunk 3]
```diff
@@ -3,9 +3,18 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: github-close-issue
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: github
+    displayName: "github close issue"
   description: |
```
Comment (sthaha) [inline github/close_issue.yaml:12]: this needs to be removed.

File: gcs/gcs-upload.yaml [hunk 4]
```diff
@@ -2,7 +2,20 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: gcs-upload
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: cloud
+    displayName: "gcs upload"
```
Comment (sthaha) [inline gcs/gcs-upload.yaml:10]: How about `Upload to GCS`

File: gcs/gcs-create-bucket.yaml [hunk 5]
```diff
@@ -2,7 +2,18 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: gcs-create-bucket
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: cloud
```
Comment (sthaha) [inline gcs/gcs-create-bucket.yaml:9]: how about `cloud, gcs` ?

File: gcs/gcs-create-bucket.yaml [hunk 6]
```diff
@@ -2,7 +2,18 @@ apiVersion: tekton.dev/v1beta1
...
 spec:
+  description: >-
+    A Task that creates a new GCS bucket.
+
+    These Tasks are for copying to and from GCS buckets from Pipelines.
```
Comment (sthaha) [inline gcs/gcs-create-bucket.yaml:14]: How about: 

> This tasks is intended to be a replacement for GCS PipelineResource and can be used for  copying to and from GCS buckets.

File: gcs/gcs-delete-bucket.yaml [hunk 7]
```diff
@@ -2,7 +2,19 @@ apiVersion: tekton.dev/v1beta1
...
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: cloud
 spec:
+  description: >-
+    A Task that deletes a GCS bucket.
+
+    These Tasks are for copying to and from GCS buckets from Pipelines.
```
Comment (sthaha) [inline gcs/gcs-delete-bucket.yaml:14]: is this correct? copy/paste error?

File: task/replace-tokens/0.1/replace-tokens.yaml [hunk 8]
```diff
@@ -2,7 +2,18 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: replace-tokens
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: tokens
+    dislayName: "replace tokens"
```
Comment (pratap0007) [inline task/replace-tokens/0.1/replace-tokens.yaml:10]: typo error -> 'displayName'

File: task/skopeo-copy/0.1/skopeo-copy.yaml [hunk 9]
```diff
@@ -2,7 +2,22 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: skopeo-copy
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: cli
+    dislayName: "skopeo copy"
```
Comment (pratap0007) [inline task/skopeo-copy/0.1/skopeo-copy.yaml:10]: typo error

File: task/upload-pypi/0.1/README.md [hunk 10]
```diff
@@ -5,7 +5,7 @@ This Task publishes Python packages to PyPI index using [Twine](https://pypi.org
 ## Install the Task
 
 ```
-kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/python/publish-package/update-pypi.yaml
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/task/upload-pypi/0.1/upload-pypi.yaml
```
Comment (pratap0007) [inline task/upload-pypi/0.1/README.md:8]: is it right v1beta1 ?

File: task/ibmcloud/0.1/README.md [hunk 11]
```diff
@@ -5,7 +5,7 @@ This task performs operations on IBM Cloud using the `ibmcloud`
 ## Install the Task
 
 ```
-kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/ibmcloud-cli/ibmcloud-cli.yaml
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/task/ibmcloud/0.1/ibmcloud.yaml
```
Comment (pratap0007) [inline task/ibmcloud/0.1/README.md:8]: can you modify  v1beta1 to master  ?

File: task/wget/0.1/wget.yaml [hunk 12]
```diff
@@ -2,7 +2,15 @@ apiVersion: tekton.dev/v1beta1
...
 metadata:
   name: wget
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: download
 spec:
+  description: >-
+    This task uses wget to download files from the internet to a workspace .
```
Comment (SM43) [inline task/wget/0.1/wget.yaml:12]: space between `workspace` and `.`

File: task/curl/0.1/curl.yaml [hunk 13]
```diff
@@ -2,7 +2,15 @@ apiVersion: tekton.dev/v1beta1
...
 metadata:
   name: curl
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: search
 spec:
+  description: >-
+    This task performs curl operation to transfer data from internet .
```
Comment (SM43) [inline task/curl/0.1/curl.yaml:12]: space between `workspace` and `.`

**Comment by nikhil-thomas** [comment]
/lgtm

I think there are some minor improvements which can be handled in later pr.
eg: tags for gcloud, ibmclous, awscli could be similar

I have a question on Task Display Name Capitalization. 
https://github.com/tektoncd/experimental/issues/581

File: conftest/conftest.yaml [hunk 14]
```diff
@@ -2,7 +2,17 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: conftest
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
```
Comment (bobcatfish) [inline conftest/conftest.yaml:8]: sweeeeet - thanks for adding the minVersion!!!

File: conftest/helm-conftest.yaml [hunk 15]
```diff
@@ -2,7 +2,18 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: helm-conftest
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tags: test
+    displayName: "helm conftest"
```
Comment (bobcatfish) [inline conftest/helm-conftest.yaml:10]: when did we decide on these tag and displayName annotations? i think we should discuss before adding and requiring these

Comment (bobcatfish) [reply conftest/helm-conftest.yaml:10]: if we do go ahead, would it make sense to preface these with tekton.dev as well?

**Comment by vdemeester** [comment]
@PuneetPunamiya can we update the description of this PR ? I think it doesn't reorganize the catalog but *just* adds the required metadata for the hub, right ?

**Comment by PuneetPunamiya** [comment]
> @PuneetPunamiya can we update the description of this PR ? I think it doesn't reorganize the catalog but _just_ adds the required metadata for the hub, right ?

Based on the proposal we are also moving tasks right to the task directory except the task which are not split

File: conftest/helm-conftest.yaml (hunk 15 above)
Comment (PuneetPunamiya) [reply conftest/helm-conftest.yaml:10]: How about we address those changes in a seperate PR ?

**Comment by bobcatfish** [comment]
I think we can keep discussing the tags over in https://github.com/tektoncd/community/pull/123 but I'm happy to merge this and #401 asap to keep the merge conflicts to a minimum

File: conftest/helm-conftest.yaml (hunk 15 above)
Comment (sthaha) [reply conftest/helm-conftest.yaml:10]: @PuneetPunamiya agree but lets create an issue before the PR is merged so that we won't miss addressing those or at least bring this up for discussion in WG meeting?

File: task/openwhisk/0.1/openwhisk.yaml [hunk 16]
```diff
@@ -3,7 +3,22 @@ apiVersion: tekton.dev/v1beta1
 kind: Task
 metadata:
   name: openwhisk
+  lables:
```
Comment (sthaha) [inline task/openwhisk/0.1/openwhisk.yaml:6]: typo labels

Comment (sthaha) [reply task/openwhisk/0.1/openwhisk.yaml:6]: We must work on a tool to validate this, otherwise these typos are quite easy to miss 
cc: @vdemeester @bobcatfish @afrittoli

**Comment by bobcatfish** [comment]
Hey @PuneetPunamiya I'm surprised to see the OWNERS file in the versioned directories, can we move them one level up?

image

I'm guessing we'll have folks owning a Task across all versions, probably not a different set for each version (tho folks could add that if they wanted to)

**Comment by sthaha** [comment]
@bobcatfish Good catch! We completely missed it!

**Comment by sthaha** [comment]
@PuneetPunamiya @bobcatfish Hopefully this fixes it - https://github.com/tektoncd/catalog/pull/443

**Comment by bobcatfish** [comment]
np @sthaha - i missed it too!! I only noticed it when I was adding my own OWNERS file :D
